Tetfuza/TetfuzaBackend/TetfuzaBackend.cs

```diff
@@ -89,70 +69,27 @@ private int DropSpeed
                 return dropFrameDelay;
             }
         }
-
-        private List<char> EmptyLine
-        {
-            get
-            {
-                return BoardLine(EMPTY_CHAR);
-            }
-        }
         
         /// <summary>
         /// Creates a new instance of a TetfuzaBackend, and initializes an empty board
         /// </summary>
-        public TetfuzaBackend()
+        public TetfuzaBackend(IConsole keyboard, IDisplay screen)
         {
-            Board = BoardInit();
-            _keyboard = new InputChecker(new SystemConsole());
+            Board = new TetfuzaBoard();
+            _keyboard = new InputChecker(keyboard);
+            _screen = screen;
         }
+
         /// <summary>
         /// Starts the game on a specified level (which determines automatic drop speed and point multiplier)
         /// </summary>
         /// <param name="startLevel"></param>
-        public TetfuzaBackend(int startLevel)
+        public TetfuzaBackend(IConsole keyboard, IDisplay screen, int startLevel)
         {
-            Board = BoardInit();
+            Board = new TetfuzaBoard();
             StartLevel = startLevel;
-            _keyboard = new InputChecker(new SystemConsole());
-        }
```

Comment:
An IConsole and IDisplay are passed into the constructor, so the backend is entirely uninvolved in the actual implementation of the objects that use these interfaces.

Reply:
That’s a great refactor.